translate_3to2.go

```diff
@@ -0,0 +1,330 @@
+			NoCreateHome:      boolV(u.NoCreateHome),
+			PrimaryGroup:      strV(u.PrimaryGroup),
+			Groups:            translateUserGroups3to2(u.Groups),
+			NoUserGroup:       boolV(u.NoUserGroup),
+			NoLogInit:         boolV(u.NoLogInit),
+			Shell:             strV(u.Shell),
+			System:            boolV(u.System),
+		})
+	}
+	return
+}
+
+func translateUserSSH3to2(in []new.SSHAuthorizedKey) (ret []old.SSHAuthorizedKey) {
+	for _, k := range in {
+		ret = append(ret, old.SSHAuthorizedKey(k))
+	}
+	return
+}
+
+func translateUserGroups3to2(in []new.Group) (ret []old.Group) {
+	for _, g := range in {
+		ret = append(ret, old.Group(g))
+	}
+	return
+}
+
+func translateGroups3to2(groups []new.PasswdGroup) (ret []old.PasswdGroup) {
+	for _, g := range groups {
+		ret = append(ret, old.PasswdGroup{
+			Name:         g.Name,
+			Gid:          g.Gid,
+			PasswordHash: strV(g.PasswordHash),
+			System:       boolV(g.System),
+		})
+	}
+	return
+}
+
+func translateUnits3to2(units []new.Unit) (ret []old.Unit) {
+	for _, u := range units {
+		ret = append(ret, old.Unit{
+			Name:     u.Name,
+			Enabled:  u.Enabled,
+			Mask:     boolV(u.Mask),
+			Contents: strV(u.Contents),
+			Dropins:  translateDropins3to2(u.Dropins),
+		})
+	}
+	return
+}
+
+func translateDropins3to2(dropins []new.Dropin) (ret []old.SystemdDropin) {
+	for _, d := range dropins {
+		ret = append(ret, old.SystemdDropin{
+			Name:     d.Name,
+			Contents: strV(d.Contents),
+		})
+	}
+	return
+}
+
+func translateDisks3to2(disks []new.Disk) (ret []old.Disk) {
+	for _, d := range disks {
+		ret = append(ret, old.Disk{
+			Device:     d.Device,
+			WipeTable:  boolV(d.WipeTable),
+			Partitions: translatePartitions3to2(d.Partitions),
+		})
+	}
+	return
+}
+
+func translatePartitions3to2(parts []new.Partition) (ret []old.Partition) {
+	for _, p := range parts {
+		ret = append(ret, old.Partition{
+			Label:    strV(p.Label),
+			Number:   p.Number,
+			Size:     intV(p.SizeMiB),
+			Start:    intV(p.StartMiB),
+			TypeGUID: strV(p.TypeGUID),
+			GUID:     strV(p.GUID),
+		})
+	}
+	return
+}
+
+func translateRaid3to2(raids []new.Raid) (ret []old.Raid) {
+	for _, r := range raids {
+		ret = append(ret, old.Raid{
+			Name:    r.Name,
+			Level:   r.Level,
+			Devices: translateDevices3to2(r.Devices),
+			Spares:  intV(r.Spares),
+			Options: translateRaidOptions3to2(r.Options),
+		})
+	}
+	return
+}
+
+func translateDevices3to2(devices []new.Device) (ret []old.Device) {
+	for _, d := range devices {
+		ret = append(ret, old.Device(d))
+	}
+	return
+}
+
+func translateRaidOptions3to2(options []new.RaidOption) (ret []old.RaidOption) {
+	for _, o := range options {
+		ret = append(ret, old.RaidOption(o))
+	}
+	return
+}
+
+func translateFilesystems3to2(fss []new.Filesystem) (ret []old.Filesystem) {
+	for _, f := range fss {
+
+		ret = append(ret, old.Filesystem{
+			// To construct a mapping for files/directories, we name the filesystem by path uniquely.
+			Name: strV(f.Path),
+			Path: f.Path,
```

Comment:
2.x path isn't the same thing. It's only used by the `root` filesystem entry, all user configs have it unset.